accepted/2021/package-validation/package-validation.md

```diff
@@ -0,0 +1,190 @@
+.NET Standard 2.0 but now they realize they want to take advantage of the new
+UTF8 string feature in .NET 7. In order to do that, they are multi-targeting for
+`netstandard2.0` and `net7.0`.
+
+Finley has written the following code:
+
+```C#
+#if NET7_0_OR_GREATER
+    public void DownloadLogs(Utf8String area)
+    {
+        DownloadLogs(area.AsSpan());
+    }
+#else
+    public void DownloadLogs(string area)
+    {
+        var utf8Bytes = Encoding.UTF8.GetBytes(area);
+        DownloadLogs(utf8Bytes);
+    }
+#endif
+
+    private void DownloadLogs(ReadOnlySpan<byte> areaUtf8)
+    {
+        // Do network call
+    }
+```
+
+When Finley builds the project it fails with the following error:
+
+> error SYSLIB1234: The method 'DownloadLogs(string)' exists for .NET Standard
+> 2.0 but does not exist for .NET 7.0. This prevents consumers who compiled
+> against .NET Standard 2.0 to run on .NET 7.0.
+
+Finley understands that they shouldn't exclude `DownloadArea(string)` but
+instead just provide an additional `DownloadArea(Utf8String)` method for .NET
+7.0 and changes the code accordingly:
+
+```C#
+#if NET7_0_OR_GREATER
+    public void DownloadLogs(Utf8String area)
+    {
+        // ...
+    }
+#endif
+
+    public void DownloadLogs(string area)
+    {
+        // ...
+    }
+```
+
+### Validation against previous version
+
+Skylar is tasked with adding support for a connection time out to their library.
+The `Connect` method looks like this right now:
+
+```C#
+public static Client Connect(string url)
+{
+    // ...
+}
+```
+
+Since a connection timeout is an advanced configuration setting they reckon they
+can just add an optional parameter:
+
+```C#
+public static Client Connect(string url, TimeSpan? timeout = default)
+{
+    // ...
+}
+```
+
+However, when they rebuild the package they are getting an error:
+
+> error SYSLIB1234: The method 'Connect(string)' exists in the previous version
+> of the NuGet package (AdventureWorks.Client 1.2.3) but no longer exists in the
+> current version being built (1.2.4). This is a breaking change.
+
+Skylar realizes that while this is not a source breaking change, it's a binary
+breaking change. They solve this problem by adding an overload instead:
+
+```C#
+public static Client Connect(string url)
+{
+    return Client(url, Timeout.InfiniteTimeSpan);
+}
+
+public static Client Connect(string url, TimeSpan timeout)
+{
+    // ...
+}
+```
+
+### Making an intentional breaking change
+
+Armani is working on version 0.7 of their library. While they don't want to make
+accidental breaking changes, they do want to make changes based on user feedback.
+
+In version 0.6 they had a `Connect()` method that accepted a timeout as an
+`int`. Users complaint that timeouts in .NET are nowadays generally modelled
+using `TimeSpan` instead. After making the necessary code change, rebuilding the
+library results in an error:
+
+> error SYSLIB1234: The method 'Connect(string, int)' exists in the previous
```

Comment:
What are we planning for diagnostic IDs? Are we planning on coming up with our own, `APC000x`? APC -> ApiCompat or `PKG000x` for package errors?

Reply:
We should discuss this in the spec review. I've added an open question in the design section